### DISCUSSION TITLE
The shared folder from macOS not shown in windows 7 virtual machine

### DISCUSSION THREAD

**Discussion by zhouhao27** [utmapp/UTM#4810] [UTM Help (macOS)]
**BEFORE SUBMITTING YOUR ISSUE, PLEASE LOOK AT THE PINNED ISSUES AND USE THE SEARCH FUNCTION TO MAKE SURE IT IS NOT ALREADY REPORTED. ALWAYS COMMENT ON AN EXISTING ISSUE INSTEAD OF MAKING A NEW ONE.**

**Describe the issue**  

I shared a folder in my host macOS 13.1. But I could not find it in my Windows 7 Virtual machine. Where to find it ? In networking?

<img width="893" alt="Screenshot 2022-12-16 at 12 27 14 PM" src="https://user-images.githubusercontent.com/8099731/208022228-4e260b36-58e5-493e-81ba-3f2a64670511.png">

<img width="693" alt="Screenshot 2022-12-16 at 12 31 17 PM" src="https://user-images.githubusercontent.com/8099731/208022656-4d3ea49f-b148-4c10-88da-e7d91a8ad6a5.png">

By the way, I installed the **Windows Guest Tool** already.

**Configuration**  
* UTM Version: 4.0.9 (68)
* macOS Version: Ventura 13.1
* Mac Chip (Intel, M1, ...):  M1

Here is the settings:

<img width="842" alt="Screenshot 2022-12-16 at 12 33 24 PM" src="https://user-images.githubusercontent.com/8099731/208022881-ca0e15ff-3ef3-481c-8227-96652a468978.png">

It mentioned to install SPICE daemon. But it didn't mention how to install and where to install?

**Comment by zhouhao27** [comment]
I even run the `map-drive.bat`:

<img width="683" alt="Screenshot 2022-12-16 at 12 43 44 PM" src="https://user-images.githubusercontent.com/8099731/208024103-3803298f-0c8c-4e0c-90bc-9340d95d865a.png">

Still can't find the shared folder.

**Reply by elewis54** [reply]
Same setup as zhouhao27.
Exactly the same result when running map-drive.bat. System error 67; network name cannot be found; operation completed successfully.

**Comment by osy** [comment]
Check that the internet is working properly in the VM. Try to browse to "http://localhost:9843/" and if it is not available, reinstall the guest tools.

**Reply by elewis54** [reply]
Internet Explorer cannot display the web page. IE can, however reach other Internet pages so I certainly have Internet connectivity. I believe the problem lies in part of the networking protocol to the host.
Same setup as zhouhao27 - M1-Pro; 16 GiB, Ventura 13.1 OS.

**Comment by elewis54** [comment]
Like the screen shots provided by @zhouhao27 Windows Explorer is only showing the local (windows machine) in the network group. I would expect that it would show the host machine in the network group. Is there a problem with network card emulation?
I have selected rtl8139 which was the default selection when setting up the windows 7 VM.
I also tried virtio-net PCI. That one did not even show the 'home network' group. FWIW, the virtio-net PCI works perfectly for file sharing with a Ubuntu 22.04 guest and a Fedora 37 guest. Both of these are 64-bit systems, however. My needs call for a 32-bit windows system.

**Comment by ronisbr** [comment]
I am seeing the same error in macOS 13.1 with Windows 7 guest.

**Comment by jremen** [comment]
Same here, tried Win 7 emuation today and can't see shared folder. Changed drivers, changed method to VirtuoFs, nothing.

MB Air M1, macOS 13.2.1 Ventura, latest UTM 4.1.5, Spice tools installed, virtio-net-pci chosen as a network card (which works fine under virtualized systems)

**Comment by JezerM** [comment]
I had this same issue when I installed the Spice Guest Tools from the UTM's ISO: no clipboard sharing, no automatic display scaling, no folder sharing.

I fixed it by reinstalling it from the original source https://www.spice-space.org/download.html (download the Windows guest binary and install it). It didn't required any modification to the VM config.

**Reply by elewis54** [reply]
Works!

To repeat and clarify:

Download the Windows installer from https://www.spice-space.org/download.html as directed.

Immediately under the "Windows binaries" heading in spice-space.org, select "spice-guest-tools". The file should download as "spice-guest-tools-latest.exe" The SHA256 sum is b5be0754802bcd7f7fe0ccdb877f8a6224ba13a2af7d84eb087a89b3b0237da2.

Proceed with installation on the Windows guest. Allow the installer to install all of the "drivers" it offers. Once complete, quit and restart Windows.

In the title bar near the right end (running Windows Full Screen will hide the title bar), select the icon entitled "shared folder". Select a Mac folder you want to share with the Windows guest.

In Windows Explorer, select "computer" on the side bar. "Spice Client (Z:)" should show under "Network Location (1)".

This is your shared folder.

**Reply by joshleaves** [reply]
Thanks! Can confirm it works too.

Also, quick reminder to future users: if you're on a Windows 7 VM, chances are you don't have proper SSL certificates to navigate the modern web, so get ready to host a quick web server on your network so you can download the file from the VM.

**Reply by Michael-Z-Freeman** [reply]
Hi. Is there a separate DL for the WebDAV file sharing now ?

**Reply by Yiak** [reply]
To the future users:
1. If you don't want to host a temp web server to download the file from the VM, you can also download it in the host and create an ISO file for the latest spice. 
2. Do not uninstall the default spice guest tool before installing the latest one, otherwise you will lose the WebDAV agent.

**Reply by stevehuang0115** [reply]
Confirm it works! Thanks all

**Reply by yasingunaydiin** [reply]
Doesnt work for me.

**Reply by GiantHardOfficial** [reply]
but what do i install on guest binary?

**Reply by boriskinay** [reply]
Confirm it works! Thanks all

**Reply by svnv-svsv-jm** [reply]
It did nothing for me... :(

**Reply by rishiktiwari** [reply]
If still not working then also download and install the [Spice WebDAV daemon](https://www.spice-space.org/download/windows/spice-webdavd/).

<img width="680" alt="screenshot" src="https://github.com/user-attachments/assets/1f4ffa12-b01d-4a26-8d41-d9fbf6e48f15">

**Reply by erezooo** [reply]
Thanks @rishiktiwari,
solved it for me: on the guest vm, I installed spice-webdavd-x64-latest.msi from the [Spice WebDAV daemon](https://www.spice-space.org/download/windows/spice-webdavd/) as @rishiktiwari mentioned. 
Then in Edge, opened http://localhost:9843/ and found my shared content

**Comment by Added-Glue** [comment]
I wasn't able to get the Shared Folder to connect automatically in Windows XP, but after all the things I tried this easy solution worked:

> All I had to do was to go to "My Network Places" and click on "Add a network place", then "Choose another network location". In the "Internet or network address field", I typed http://localhost:9843/
> Bingo!

[XP: spice-webdavd failed to start: "privileges to start system service"](https://gitlab.gnome.org/GNOME/phodav/-/issues/19#note_1434113])

**Reply by ag14spirit** [reply]
@Added-Glue thank you, this one solved the issue for me!

**Comment by digitalghost-dev** [comment]
I was able to get spice tools installed by following these [instructions](https://docs.getutm.app/guest-support/windows/) and clicking on the Shared Folder icon after.
